src/components/ScholarshipAmountField.jsx

```diff
@@ -24,55 +25,61 @@ const useStyles = makeStyles(() => ({
 
 function ScholarshipAmountField(props) {
   const classes = useStyles();
-  const {
-    helperText,
-    amountType,
-    minAmount,
-    maxAmount,
-    onTypeChange,
-    updateAmount,
-    labelStyle,
-  } = props;
+  const { labelStyle, formik } = props;
 
-  const isRange = amountType === AmountType.Range;
-  const isFixed = amountType === AmountType.Fixed;
-  const error = !!helperText; // no error if helperText empty
+  function updateAmount(min, max) {
+    formik.setFieldValue('amount.min', min, true);
+    formik.setFieldValue('amount.max', max, true);
+  }
+
+  function onTypeChange(e) {
+    formik.setFieldValue('amount.type', e.target.value, true);
+  }
+
+  const amountType = formik.values.amount.type;
+  const minAmount = formik.values.amount.min;
+  const maxAmount = formik.values.amount.max;
+  const minError = getIn(formik.errors, 'amount.min');
+  const maxError = getIn(formik.errors, 'amount.max');
+
+  function formatHelperText() {
+    if (!minError && !maxError) return;
+    if (getIn(formik.touched, 'amount.min') && minError) return minError;
+    if (getIn(formik.touche, 'amount.max') && maxError) return maxError;
+  }
 
   const inputFields = {};
-  inputFields[AmountType.Range] = (
+  inputFields[AmountType.Varies] = (
     <>
       <AmountTextField
-        error={isRange && error && (!!maxAmount || minAmount >= maxAmount)}
+        error={Boolean(minError)}
         value={minAmount || ''}
         onChange={(e) => {
           const val = parseInt(e.target.value, 10);
           updateAmount(val || 0, maxAmount);
         }}
-        disabled={!isRange}
       />
 
       <RemoveIcon className={classes.dash} />
 
       <AmountTextField
-        error={isRange && error && !minAmount && !maxAmount}
+        error={Boolean(maxError)}
         value={maxAmount || ''}
         onChange={(e) => {
           const val = parseInt(e.target.value, 10);
           updateAmount(minAmount, val || 0);
         }}
-        disabled={!isRange}
       />
     </>
   );
   inputFields[AmountType.Fixed] = (
     <AmountTextField
-      error={isFixed && error}
+      error={getIn(formik.touched, 'amount.min') && Boolean(minError)}
```

Comment:
without this, the field will get highlighted as an error even when the input is correct. I am planning to change when the validation runs to hopefully avoid these checks.